### PR TITLE
Cats 1661 refactor receipt form to submit using ajax.

### PR DESCRIPTION
**Pull request by dawitd**

### CODE REVIEW

File: app/views/receipts/_form.html.erb [hunk 1]
```diff
@@ -241,43 +269,52 @@
...
+
+         
+
       </div>
     </div>
 
   </div>
   <% end %>
 
   <script type="text/javascript">
```
Comment (yareda) [inline app/views/receipts/_form.html.erb:288]: Can we move script below this line to corresponding *.js file? This view is very cluttered and difficult to navigate through.

Comment (yareda) [reply app/views/receipts/_form.html.erb:288]: @dawitd check if moving to it's own *.js file a better option.

File: app/views/receipts/_form.html.erb [hunk 2]
```diff
@@ -316,17 +358,113 @@
         });
       });
 
-      $('#submit-receipt').click(function (e) {
+      $('#submit-receipt-and-new').click(function (e) {
 
         if ($('.receipt-line').length === 0) {
-          toastr.error("You need to atleast add one commodity.");
+          toastr.error("You need to at least add one commodity.");
```
Comment (yareda) [inline app/views/receipts/_form.html.erb]: @dawitd rephrase error message to -> "You are required to add at one commodity to receive"

**Review by yareda — CHANGES_REQUESTED** [review]
@dawitd Not sure why we need to change the form to submit over ajax call!

File: app/views/receipts/_form.html.erb (hunk 1 above)
Comment (dawitd) [reply app/views/receipts/_form.html.erb:288]: I have moved the pure js scripts to a separate file - receipt.js file. we can not move all scripts to a js file as there are server side scripts.

**Comment by yareda** [comment]
@nathnael please provide your review and approval

**Comment by nathnael** [comment]
@dawitd 
Reviews:
1. Unit conversion not handled
2. Received value is negative
3. When I receive "Received" decreased & "Remaining" increased. May be it's because the "Received" value is negative

**Comment by dawitd** [comment]
In testing the unit issue - make sure that you have updated uom_type and ration fields of the unit_of_measurement table in relation to the reference unit.

**Review by nathnael — APPROVED** [review]

**Review by yareda — APPROVED** [review]

**Pull request merged**